### ISSUE TITLE
Release a new stable version of all libraries (3.0)

### ISSUE THREAD

**Issue by addshore** [addwiki/addwiki#23]
From https://phabricator.wikimedia.org/T251344

**Comment by samwilson** [comment]
I'm running into an issue where I was requiring `"addwiki/mediawiki-api-base": "^2.5"` and it's now failing because the required PHP version jumped between 2.7 and 2.8 (to PHP 7.3). Am I right in thinking that this is a breaking change and so should waited till 3.0?

**Comment by addshore** [comment]
> I'm running into an issue where I was requiring `"addwiki/mediawiki-api-base": "^2.5"` and it's now failing because the required PHP version jumped between 2.7 and 2.8 (to PHP 7.3). Am I right in thinking that this is a breaking change and so should waited till 3.0?

https://medium.com/@sampart/semantic-versioning-when-you-change-the-required-programming-language-version-16a3a3555c95

TLDR; the interfaces provided by the libraries remain the same so this was not added as a major release.
Which version are you running with? (I change the allowed versions to be any currently released version of php that everything is also tested with.
It probably runs on any version of 7.0

The diff between 2.7 and 2.8 can be found here https://github.com/addwiki/addwiki/compare/2.7.0...2.8.0

**Comment by samwilson** [comment]
Ah cool, thanks for the link.

I was trying to install it with PHP 7.2 (and 7.3 and 7.4), and have now [set it](https://github.com/samwilson/basildon/blob/master/composer.json) to `"addwiki/mediawiki-api-base": "2.7.*"` and it's fine.

I guess I always set the PHP requirement of a library to be the *minimum* possible, rather than the minimum supported, and only increment it when the library wants to use a newer feature. Not complaining though! It's great that you're back working on these libraries. :-)

**Comment by addshore** [comment]
So if you try to install the libraries with PHP7.2, you can still declare a dependency on ~2.0, but if you specifically need features in 2.8 then you'd need to use PHP7.3 or above.
Looking back I don't see anything noticeable that specifically would make 2.8 not work with 7.2, but I feel that I removed the 7.2 CI for a reason (probably because something broke), can't be sure now!

3.0 will require 7.3+ again.
At some point I'll make a 8.0 only release too!
